### PR TITLE
tests: Scaffold the Charmhub workflow test

### PR DESCRIPTION
**Pull request by lengau**
- [x] Have you followed the guidelines for contributing?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] Have you successfully run `tox`?

-----

An early version of a full charm workflow integration test. I'm writing it to help test #124, but these initial steps don't require any of that code and still test things our existing tests don't, so I thought it worth bringing on its own.

### CODE REVIEW

File: tests/integration/charmhub/test_charmhub_workflow.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2021-2022 Canonical Ltd.
```
Comment (tigarmo) [inline tests/integration/charmhub/test_charmhub_workflow.py:3]:
```suggestion
# Copyright 2023 Canonical Ltd.
```

**Review by tigarmo — APPROVED** [review]
looks good! Thanks

File: tests/integration/charmhub/test_charmhub_workflow.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+import time
+
+from craft_store.models import revisions_model
+
+from tests.integration.conftest import needs_charmhub_credentials
+
+
+@needs_charmhub_credentials()
+def test_full_charm_workflow(charm_client, charmhub_charm_name, fake_charms):
+    """A workflow test for uploading a full charm.
```
Comment (mr-cal) [inline tests/integration/charmhub/test_charmhub_workflow.py:26]: What's a full charm?  A charm with resources?

**Review by mr-cal — APPROVED** [review]

File: tests/integration/charmhub/test_charmhub_workflow.py (hunk 2 above)
Comment (lengau) [reply tests/integration/charmhub/test_charmhub_workflow.py]: No, just "full" in the wrong space. It should be "A full workflow chest for uploading a charm." (Fixed in the latest commit)

**Pull request merged**